### ISSUE TITLE
Consider using unknown instead of any for AxiosResponse<T> default

### ISSUE THREAD

**Issue by JeremyLoy** [axios/axios#3314]
#### Is your feature request related to a problem? Please describe.

https://www.typescriptlang.org/docs/handbook/basic-types.html#unknown

TL;DR unknown is the type safe equivalent of any.

When getting data from AxiosResponse, the caller should be forced to make assertions about the response type. Using any can cause subtle programming errors. These errors won't be caught by most linter configurations, as they can only enforce the absence of the any type from first party code.

This would be a breaking change, so it might have to be saved for Axios v1.

#### Describe the solution you'd like

Type safety to be enforced more rigorously by Axios.

#### Describe alternatives you've considered

Creating a linting rule specifically for Axios.

**Comment by jasonsaayman** [comment]
In version 2 we will be going fully TS in a large upgrade, closing as this is not relevant anymore as the current versions will not be changed. Thanks for this
